core/src/main/scala/kafka/controller/TopicDeletionManager.scala

```diff
@@ -88,6 +86,10 @@ class TopicDeletionManager(controller: KafkaController,
   val deleteTopicStateChanged: AtomicBoolean = new AtomicBoolean(false)
   var deleteTopicsThread: DeleteTopicsThread = null
   val isDeleteTopicEnabled = controller.config.deleteTopicEnable
+  val topicsToBeDeleted: mutable.Set[String] = mutable.Set.empty[String]
+  if (isDeleteTopicEnabled)
+    topicsToBeDeleted ++= initialTopicsToBeDeleted
+  val partitionsToBeDeleted: mutable.Set[TopicAndPartition] = topicsToBeDeleted.flatMap(controllerContext.partitionsForTopic)
```

Comment:
Yes, that would happen in the DeleteTopicListener in this patch. When a new controller is elected, the DeleteTopicListener will check if the deleteTopic flag is enabled and if it is disabled, it will remove the clear the topics under the deleteTopic path.